.gitignore

```diff
@@ -34,6 +34,9 @@ html/
 *.so
 *.dylib
 
+# exceptions for AR
+!libarcore*.so
```

Comment:
It may be worth keeping the top-level `.gitignore` minmal, and add two specific `.gitignore` files in the uitools and AR folders if they have different requirements for what needs ignored.

Reply:
For now I would like to have a single .gitignore for the entire repo at the root. We've never needed to have different ones, so that sounds like a potential pitfall keeping them in sync, or figuring out why changes to the root .gitignore aren't working (since they're being overridden by a separate one).

If we need more exceptions in the future we can start splitting them out.